Bookmark-Manager/src/test/java/com/bookmark/bookmarks/BookmarkServiceTest.java

```diff
@@ -0,0 +1,215 @@
+  FolderRepository folderRepository = mock(FolderRepository.class);
+  BookmarkService bookmarkService = new BookmarkService(bookmarkRepository, folderRepository);
+
+
+
+  @Test
+  void test_create_bookmark_with_valid_data() {
+    UUID folderId = UUID.randomUUID();
+    Bookmark bookmark = new Bookmark();
+    bookmark.setTitle("Sample Title");
+    bookmark.setUrl("http://example.com");
+    bookmark.setUserId(UUID.randomUUID());
+    bookmark.setFolderId(folderId);
+
+    when(folderRepository.findById(folderId)).thenReturn(Optional.of(new Folder()));
+    when(bookmarkRepository.save(any(Bookmark.class))).thenReturn(bookmark);
+
+    Bookmark savedBookmark = bookmarkService.createBookmark(bookmark);
+
+    assertNotNull(savedBookmark);
+    assertEquals("Sample Title", savedBookmark.getTitle());
+    assertEquals("http://example.com", savedBookmark.getUrl());
+  }
+
+    @Test
+    void test_create_bookmark_with_invalid_data() {
+        Bookmark bookmark = new Bookmark();
+
+        try {
+            bookmarkService.createBookmark(bookmark);
+        } catch (BookmarkManagerException e) {
+            assertEquals("Invalid Arguments", e.getMessage());
+        }
+    }
+
+    @Test
+    void test_create_bookmark_with_invalid_folder_id() {
+        Bookmark bookmark = new Bookmark();
+        bookmark.setTitle("Sample Title");
+        bookmark.setUrl("http://example.com");
+        bookmark.setUserId(UUID.randomUUID());
+        bookmark.setFolderId(UUID.randomUUID());
+
+        when(folderRepository.findById(bookmark.getFolderId())).thenReturn(Optional.empty());
+
+        try {
+            bookmarkService.createBookmark(bookmark);
+        } catch (IllegalArgumentException e) {
+            assertEquals("Invalid folder ID", e.getMessage());
+        }
+    }
+
+    @Test
+    void test_get_bookmark_by_id() {
+        UUID bookmarkId = UUID.randomUUID();
+        Bookmark bookmark = new Bookmark();
+        bookmark.setId(bookmarkId);
+        bookmark.setTitle("Sample Title");
+        bookmark.setUrl("http://example.com");
+        bookmark.setUserId(UUID.randomUUID());
+
+        when(bookmarkRepository.findById(bookmarkId)).thenReturn(Optional.of(bookmark));
+
+        Bookmark savedBookmark = bookmarkService.getBookmarkById(bookmarkId);
+
+        assertNotNull(savedBookmark);
+        assertEquals(bookmarkId, savedBookmark.getId());
+        assertEquals("Sample Title", savedBookmark.getTitle());
+        assertEquals("http://example.com", savedBookmark.getUrl());
+    }
+
+    @Test
+    void test_get_bookmark_by_id_with_invalid_id() {
+        UUID bookmarkId = UUID.randomUUID();
+
+        when(bookmarkRepository.findById(bookmarkId)).thenReturn(Optional.empty());
+
+        try {
+            bookmarkService.getBookmarkById(bookmarkId);
+        } catch (BookmarkNotFoundException e) {
+          assertNotNull(e.getMessage());
+        }
+    }
+
+    @Test
+    void test_update_bookmark() {
```

Comment:
Also use "Given-When-Then" style pattern for better readability 

For example:
```
// Given-When-Then style
@Test
void shouldUpdateBookmarkWhenValidIdProvided() {
    // Given
    UUID bookmarkId = UUID.randomUUID();
    Bookmark existingBookmark = new Bookmark();
    existingBookmark.setTitle("Old Title");
    
    Bookmark updatedBookmark = new Bookmark();
    updatedBookmark.setTitle("New Title");
    
    when(bookmarkRepository.findById(bookmarkId)).thenReturn(Optional.of(existingBookmark));
    when(bookmarkRepository.save(any())).thenReturn(updatedBookmark);
    
    // When
    Bookmark result = bookmarkService.updateBookmark(bookmarkId, updatedBookmark);
    
    // Then
    assertEquals("New Title", result.getTitle());
    verify(bookmarkRepository).findById(bookmarkId);
    verify(bookmarkRepository).save(any());
}

Reply:
I have spereater without comment will add // Given-When-Then styles